active/RowColumnAPIChange/rowcolumnapichange.md

```diff
@@ -0,0 +1,168 @@
+
+# Examples
+<!-- Use this section to explain the features of the API, showing
+example code with each description. The general format is: 
+  feature explanation,
+  example code
+  feature explanation,
+  example code
+  etc.-->
+  
+<!-- Code samples should be in C# and/or C++/WinRT -->
+
+<!-- As an example of this section, see the Examples section for the PasswordBox control 
+(https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box#examples). -->
+Current syntax to define a Grid with one column in code-behind:
+```csharp
+var grid = new Grid();
+var column = new ColumnDefinition();
+column.Width = new GridLength(1.0, GridUnitType.Star);
+grid.ColumnDefinitions.Add(column);
+```
+
+or
+
+```csharp
+var column = new ColumnDefinition
+{
+    Width = new GridLength(1.0, GridUnitType.Star)
+};
+```
+
+
+New syntax to define a Grid with one column in code-behind:
+```csharp
+var grid = new Grid();
+grid.ColumnDefinitions.Add(new ColumnDefinition(1.0, GridUnitType.Star)); // Width="1*"
+```
+
+New syntax to define a Grid with one simple column in code-behind, where the GridUnitType is a pixel:
+```csharp
+grid.ColumnDefinitions.Add(new ColumnDefinition(500.0)); // Width="500px"
+```
+
+<!-- # Remarks -->
+<!-- Explanation and guidance that doesn't fit into the Examples section. -->
+
+<!-- APIs should only throw exceptions in exceptional conditions; basically,
+only when there's a bug in the caller, such as argument exception.  But if for some
+reason it's necessary for a caller to catch an exception from an API, call that
+out with an explanation either here or in the Examples -->
+
+# API Notes
+ColumnDefinition constructor will be overloaded to take a double `pixelWidth` OR take two arguments consisting of a double `width` and a GridUnitType `type`:
+
+```csharp
+public ColumnDefinition(double pixelWidth) { ... }
+```
+
+```csharp
+public ColumnDefinition(double width, GridUnitType type) { ... }
+```
+
+RowDefinition constructor will be overloaded to take a double `pixelHeight` OR take two arguments consisting of a double `height` and a GridUnitType `type`:
+
+```csharp
+public RowDefinition(double pixelHeight) { ... }
+```
+
+```csharp
+public RowDefinition(double height, GridUnitType type) { ... }
+```
+
+# API Details
+```csharp
+//RowDefinition API:
+[contract(Windows.Foundation.UniversalApiContract, 1)]
+[webhosthidden]
+[static_name("Windows.UI.Xaml.Controls.IRowDefinitionStatics", 5adf3fe5-2056-4724-94d6-e4812b022ec8)]
+[interface_name("Windows.UI.Xaml.Controls.IRowDefinition", 4abae829-d80c-4a5e-a48c-f8b3d3b6533d)]
+runtimeclass RowDefinition
+    : Windows.UI.Xaml.DependencyObject
+{
+    // vvvv
+    RowDefinition(double height, GridUnitType type);
```

Comment:
A few questions:
1. If there is a difference between "\*" and "\*1" how are users able to set "\*"? Should they set `height` to `0`?
2. What about `GridUnitType.Auto`? Will height be ignored in that case? It has to essentially right?
3. Out of curiosity should `Double.NaN` be considered an error for `GridUnitType.Pixel` and `GridUnitType.Star` or should we do some special logic for that?

Or are this all topics that are just handled by GridLength and we should just pass the arguments and do no checking at all?

Reply:
My assumption on behavior based on current behavior:
1: Setting it to 0* would give it 0 height. You should set the number to the fractions you want the height to be. Same as it works in Xaml.
2. We already have the static `GridLength.Auto` getter (unless you use the WinUI3 alpha where it's oddly missing). If you use the type-overload, the number won't really matter.
3. I'd vote for an error here. Perhaps `Auto` could allow `NaN`, but would prefer consistency.

Reply:
Thanks for bringing these things up! I'm not too sure if these are handled by GridLength, but I'll specify these special cases in the spec remarks. 

Good catch with GridLength.Auto - I agree that the number should be ignored if you specify Auto as your GridUnitType.

As for NaN, I think an error would be most appropriate for all GridUnitTypes - if someone is calculating a NaN value for their layout, it would seem that there's some kind of larger underlying issue.